docs: introduce docs on how to deploy to the platform, and help page for the licensing tool (#2479)
TimefoldAI/timefold-solver: docs/src/modules/ROOT/pages/deploying-to-platform/metrics.adoc

* *Solve graphs*: output metrics can be plotted alongside the hard, medium, and soft score as the solver runs, so consumers see how a metric evolves over time, not just its final value.
See https://docs.timefold.ai/timefold-platform/latest/how-tos/interpreting-model-run-results#_solve_graphs[Solve graphs] for details.
* *Dataset comparison*: input and output metrics are available as columns when comparing multiple datasets side by side, making it possible to spot how problem size or solution quality differs between them.
See https://docs.timefold.ai/timefold-platform/latest/how-tos/comparing-runs[Comparing datasets] for details.
* *Insights*: both metric types can be tracked over time across many datasets, so consumers can tell whether operational planning is improving or degrading.
See https://docs.timefold.ai/timefold-platform/latest/how-tos/insights[Insights] for details.

Without input and output metrics, none of these views have anything model-specific to show beyond the score.

See xref:running-timefold-solver/service/exposing-metrics.adoc[Exposing metrics] for how to define and implement input and output metrics on your model.

> While you're at it, give each metric a proper OpenAPI `title`, `description`, and `format` in its `@Schema` annotation.

Reviewer: ```suggestion
You should also give each metric an OpenAPI `title`, `description`, and `format` in its `@Schema` annotation.
```
"While you're at it," is super informal.
